dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -565,7 +575,31 @@
             activeTask = new IntervalTask(this);
             timer.schedule(activeTask, interval, interval);
         }
+        /**
+         * if coreThreads has been configured to a small value, we will use the
+         * configured value as guidance for how large to make poolSize changes
+         */
+        if (coreThreads < NUMBER_CPUS * 2) {
+            poolChangeBasis = coreThreads / 2;
```

Comment:
Do we want to have a poolChangeBasis of 1?  Is that too small?  Do we want a minimum of 4 or something?

Reply:
If the operator has configured coreThreads=1, then I think we should take that as an indicator that the expected best poolsize will be quite small ... the general principle is to respect the user's config, if present, and if no config is present, follow the general rules (based on NUMBER_CPUS).

Reply:
That bothers me.  We at times set thread pools to a minimum of 1 since that is the true minimum and a maximum of a large amount.  Increasing by only 1 has been an issue in the past and I think it would be a bad idea to go back to it.  We may want to do some runs with a minimum of 1 and see how things behave to see if this is the right answer.  Testing both the hang scenario and the large scenarios with different workload times.

Reply:
Mmmmm ... I don't really understand 'minimum of 1 since that is the true minimum' ... 
The default assumption for Liberty is that the poolSize will be completely autonomously controlled, with no config required by the user. In that situation, the pool did not self-size very well for certain workloads, which is why we changed the default adjustment size (among other parts of the algorithm). Our new default adjustment size is half of the default coreThreads size (which is also NUMBER_CPUS).
If the operator sets coreThreads to a small value, it seems to be a signal that the default coreThreads setting (2*NUMBER_CPUS) is not working well for their config/app/environment. If the optimal coreThreads setting is 1 or 2, then the optimal total threads to run the app must be pretty small too, so small poolSize adjustments make sense.
Of course, I agree that we should test this ... but if we have for example 16 h/w threads and set coreThreads=1, it will inevitably take longer to get to max throughput than it did before. 
The controller should still do a pretty good job of adjusting to reality as it discovers it, I expect ... if you recall the very first prototype involved just expanding the compareRange from +/-1 to a wider view of the historical data, without changing the poolIncrement/decrement, and that worked quite well, though when the optimal poolsize was many 100's of threads it took a while to get there. 
Also, now the controller has markers to double the increment size when the poolSize passes [16 * poolChangeBasis] (so at 16 threads in this case), and double it again at [64 * poolChangeBasis], so we will not be stuck at single-thread increments if we keep finding better perf at larger poolsizes. 
And the hangResolution scenario will work much quicker than before, because of the reduced controller cycle time (hangInterval) when it enters hang mode.
So if coreThreads is set to 1 or 2  *when it really shouldn't be* then it won't work as well as if coreThreads was left to its default, but it will still work a lot better than the prior controller worked in that case ... and since we cannot know whether the setting is optimal or not,  if coreThreads is set to 1 or 2, I think we need to honor the implicit intent of that setting, which seems to be 'not so many threads, please'.
An example of where this setting would be correct/helpful, because of stuff in the environment that the controller cannot know about, is Mel's Power9 test config, where he was running 40 JVMs in the single machine with 192 h/w threads. In that case, the optimal coreThread count is a lot less than [2 * h/w threads], and the optimal poolSize adjustment is proportionally a lot smaller. 
Sorry for so many words ... just trying to fully explain my thinking on this item. :-)